omod/src/main/java/org/openmrs/module/shr/rest/web/controller/EncounterController.java

```diff
@@ -94,12 +83,19 @@ public Object postDocument(
 			Provider provider = getOrCreateProvider(providerId, providerIdType);
 			EncounterRole role = getDefaultEncounterRole();
 			EncounterType type = getOrCreateEncounterType(encounterType);
+			String encounterUUID = role.getUuid();
 			boolean url = (isURL!=null && "true".equalsIgnoreCase(isURL)) ? true : false;
-			Content content = buildContent(contentType, request, formatCode, url);
+			CodedValue typeCode = new CodedValue(typeCode_code, typeCode_codingScheme, typeCode_codingScheme);
+			CodedValue formatCode = new CodedValue(formatCode_code, formatCode_codingScheme, formatCode_codingScheme);
+
+			Content content = buildContent(encounterUUID, contentType, request, typeCode, formatCode, url);
```

Comment:
Why are passing the encounter role type id here? Isn't this where uniqueID should be used? If it's not provided, we can use UUID.randomUUID()